### PR TITLE
added sql db ledgerDigestStorageEnabled

### PR DESCRIPTION
**Pull request by muzzamilinovaqo**

### CODE REVIEW

File: exports.js [hunk 1]
```diff
@@ -989,7 +990,7 @@ module.exports = {
         'eventHubMinimumTLSversion'     : require(__dirname + '/plugins/azure/eventhub/eventHubMinimumTLSversion.js'),
 
         'accessLogsEnabled'             : require(__dirname + '/plugins/azure/frontdoor/accessLogsEnabled.js'),
-        'frontDoorMinimumTlsVersion'    : require(__dirname + '/plugins/azure/frontdoor/frontDoorMinimumTlsVersion.js')
+        'frontDoorMinimumTlsVersion'    : require(__dirname + '/plugins/azure/frontdoor/frontDoorMinimumTlsVersion.js'),
```
Comment (alphadev4) [inline exports.js:993]:
```suggestion
        'frontDoorMinimumTlsVersion'    : require(__dirname + '/plugins/azure/frontdoor/frontDoorMinimumTlsVersion.js')
```

File: plugins/azure/sqldatabases/dbLedgerDigestStorageEnabled.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+var helpers = require('../../../helpers/azure');
+
+module.exports = {
+    title: 'Ledger Digest Storage Enabled',
+    category: 'SQL Databases',
+    domain: 'Databases',
+    description: 'Ensure automatic Ledger digest storage is enabled.',
+    more_info: 'Configuring automatic Ledger digest storage allows for the generation and storage of digests for later verification.',
+    recommended_action: 'Configure an Azure Storage account or Azure Confidential Ledger for automatic Ledger digest storage. ',
+    link: 'https://docs.microsoft.com/en-us/azure/sql-database/sql-database-ledger-overview',
```
Comment (alphadev4) [inline plugins/azure/sqldatabases/dbLedgerDigestStorageEnabled.js:11]:
```suggestion
    link: 'https://learn.microsoft.com/en-us/sql/relational-databases/security/ledger/ledger-overview?view=sql-server-ver16',
```

File: plugins/azure/sqldatabases/dbLedgerDigestStorageEnabled.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+                if (!databases || databases.err || !databases.data) {
+                    helpers.addResult(results, 3,
+                        'Unable to query for SQL server databases: ' + helpers.addError(databases), location, server.id);
+                } else {
+                    if (!databases.data.length) {
+                        helpers.addResult(results, 0,
+                            'No databases found for SQL server', location, server.id);
+                    } else {
+                        databases.data.forEach(database=> {
```
Comment (alphadev4) [inline plugins/azure/sqldatabases/dbLedgerDigestStorageEnabled.js:47]:
```suggestion
                        databases.data.forEach(database => {
```

File: plugins/azure/sqldatabases/dbLedgerDigestStorageEnabled.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+                            var ledgerDigestUploads = helpers.addSource(cache, source, ['ledgerDigestUploads', 'list', location, database.id]);
+                            if (!ledgerDigestUploads || ledgerDigestUploads.err || !ledgerDigestUploads.data) {
+                                helpers.addResult(results, 3, 'Unable to query for Azure ledger: ' + helpers.addError(ledgerDigestUploads), location, database.id);
```
Comment (alphadev4) [inline plugins/azure/sqldatabases/dbLedgerDigestStorageEnabled.js:50]:
```suggestion
                                helpers.addResult(results, 3, 'Unable to query for Legder Digest Uploads for SQL servers: ' + helpers.addError(ledgerDigestUploads), location, database.id);
```

File: plugins/azure/sqldatabases/dbLedgerDigestStorageEnabled.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+                            } else {
+                                if (ledgerDigestUploads.data.length && ledgerDigestUploads.data[0].state.toLowerCase() == 'enabled') {
+                                    helpers.addResult(results, 0, 'Automatic Ledger digest storage is enabled for SQL database', location, database.id);
```
Comment (alphadev4) [inline plugins/azure/sqldatabases/dbLedgerDigestStorageEnabled.js:53]:
```suggestion
                                    helpers.addResult(results, 0, 'Ledger automatic digest storage is enabled for SQL database', location, database.id);
```

File: plugins/azure/sqldatabases/dbLedgerDigestStorageEnabled.js [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+                                } else {
+                                    helpers.addResult(results, 2, 'Automatic Ledger digest storage is not enabled for SQL database', location, database.id);
```
Comment (alphadev4) [inline plugins/azure/sqldatabases/dbLedgerDigestStorageEnabled.js:55]:
```suggestion
                                    helpers.addResult(results, 2, 'Ledger automatic digest storage is not enabled for SQL database', location, database.id);
```

File: plugins/azure/sqldatabases/dbLedgerDigestStorageEnabled.spec.js [hunk 7]
```diff
@@ -0,0 +1,196 @@
...
+            );
+
+            enableAutomaticLedgerDigestStorage.run(cache, {}, callback);
+        });
+
+        it('should give passing result if Automatic Ledger digest storage is enabled for the database', function(done) {
+            const callback = (err, results) => {
+                expect(results.length).to.equal(1);
+                expect(results[0].status).to.equal(0);
+                expect(results[0].message).to.include('Automatic Ledger digest storage is enabled for SQL database');
```
Comment (alphadev4) [inline plugins/azure/sqldatabases/dbLedgerDigestStorageEnabled.spec.js:104]:
```suggestion
                expect(results[0].message).to.include('Ledger automatic digest storage is enabled for SQL database');
```

File: plugins/azure/sqldatabases/dbLedgerDigestStorageEnabled.spec.js [hunk 8]
```diff
@@ -0,0 +1,196 @@
...
+            );
+
+            enableAutomaticLedgerDigestStorage.run(cache, {}, callback);
+        });
+
+        it('should give failing result if Automatic Ledger digest storage is disabled for the database', function(done) {
+            const callback = (err, results) => {
+                expect(results.length).to.equal(1);
+                expect(results[0].status).to.equal(2);
+                expect(results[0].message).to.include('Automatic Ledger digest storage is not enabled for SQL database');
```
Comment (alphadev4) [inline plugins/azure/sqldatabases/dbLedgerDigestStorageEnabled.spec.js:122]:
```suggestion
                expect(results[0].message).to.include('Ledger automatic digest storage is not enabled for SQL database');
```

File: plugins/azure/sqldatabases/dbLedgerDigestStorageEnabled.spec.js [hunk 9]
```diff
@@ -0,0 +1,196 @@
...
+            );
+
+            enableAutomaticLedgerDigestStorage.run(cache, {}, callback);
+        });
+
+        it('should give unknown result if unable to query for Azure ledger', function(done) {
+            const callback = (err, results) => {
+                expect(results.length).to.equal(1);
+                expect(results[0].status).to.equal(3);
+                expect(results[0].message).to.include('Unable to query for Azure ledger');
```
Comment (alphadev4) [inline plugins/azure/sqldatabases/dbLedgerDigestStorageEnabled.spec.js:179]:
```suggestion
                expect(results[0].message).to.include('Unable to query for Legder Digest Uploads for SQL database:');
```

File: plugins/azure/sqldatabases/dbLedgerDigestStorageEnabled.js [hunk 10]
```diff
@@ -0,0 +1,69 @@
+var async = require('async');
+var helpers = require('../../../helpers/azure');
+
+module.exports = {
+    title: 'Ledger Digest Storage Enabled',
+    category: 'SQL Databases',
+    domain: 'Databases',
+    description: 'Ensure that ledger automatic digest storage is enabled.',
+    more_info: 'Configuring automatic Ledger digest storage allows for the generation and storage of digests for later verification.',
```
Comment (alphadev4) [inline plugins/azure/sqldatabases/dbLedgerDigestStorageEnabled.js:9]:
```suggestion
    more_info: 'Configuring Ledger automatic digest storage allows the generation and storage of digests for later verification. Automatic database digests are generated on a predefined interval of 30 seconds and uploaded to the selected storage service.',
```

File: plugins/azure/sqldatabases/dbLedgerDigestStorageEnabled.js [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+    recommended_action: 'Configure an Azure Storage account or Azure Confidential Ledger for automatic Ledger digest storage. ',
```
Comment (alphadev4) [inline plugins/azure/sqldatabases/dbLedgerDigestStorageEnabled.js:10]:
```suggestion
    recommended_action: 'Modify SQL database ledger and add storage account for automatic digest storage.',
```

File: plugins/azure/sqldatabases/dbLedgerDigestStorageEnabled.js [hunk 12]
```diff
@@ -0,0 +1,69 @@
+var async = require('async');
+var helpers = require('../../../helpers/azure');
+
+module.exports = {
+    title: 'Ledger Digest Storage Enabled',
```
[the remark this answers is not in the document]
Comment (muzzamilinovaqo) [reply plugins/azure/sqldatabases/dbLedgerDigestStorageEnabled.js:5]: it's a feature of ledger that's why. Let me know if you still want me to change it.

**Review by mehakseedat63 — CHANGES_REQUESTED** [review]

**Review by mehakseedat63 — APPROVED** [review]

**Pull request merged**